src/optimizer/stats/stats_storage.cpp

```diff
@@ -109,20 +97,21 @@ void StatsStorage::InsertOrUpdateColumnStats(
   LOG_TRACE("InsertOrUpdateColumnStats, %d, %lf, %lf, %s, %s, %s", num_rows,
             cardinality, frac_null, most_common_vals.c_str(),
             most_common_freqs.c_str(), histogram_bounds.c_str());
-  auto column_stats_catalog = catalog::ColumnStatsCatalog::GetInstance(nullptr);
+  auto pg_column_stats = catalog::Catalog::GetInstance()
+                             ->GetSystemCatalogs(database_id)
+                             ->GetColumnStatsCatalog();
   auto &txn_manager = concurrency::TransactionManagerFactory::GetInstance();
 
   bool single_statement_txn = false;
   if (txn == nullptr) {
     single_statement_txn = true;
     txn = txn_manager.BeginTransaction();
   }
-  column_stats_catalog->DeleteColumnStats(database_id, table_id, column_id,
-                                          txn);
-  column_stats_catalog->InsertColumnStats(
-      database_id, table_id, column_id, num_rows, cardinality, frac_null,
-      most_common_vals, most_common_freqs, histogram_bounds, column_name,
-      has_index, pool_.get(), txn);
+  pg_column_stats->DeleteColumnStats(table_id, column_id, txn);
```

Comment:
DeleteColumnStats if they exist, I assume. May be helpful to add comment stating that, if correct.